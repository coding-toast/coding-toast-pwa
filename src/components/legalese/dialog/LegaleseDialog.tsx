import * as React from 'react';
import { useState } from 'react';
import { Button, Modal } from 'react-bootstrap';
import { ContractArea } from 'components';

interface ILegaleseDialogProps {
  muted?: boolean;
  title: string;
}

const LegaleseDialog: React.FC<ILegaleseDialogProps> = (props) => {
  const { title, muted, children } = props;

  const [show, setShow] = useState(false);

  const classes = muted ? 'text-muted' : '';

  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  return (
    <>
      <Button variant='link' onClick={handleShow} size='sm' className={`p-0 ${classes}`}>
        {title}
      </Button>

      <Modal show={show} size='lg' onHide={handleClose}>
        <Modal.Header closeButton>
          <Modal.Title>{title}</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <ContractArea>{children}</ContractArea>
        </Modal.Body>
        <Modal.Footer>
          <Button onClick={handleClose} className='btn-accent-secondary' block>
            I understand
          </Button>
        </Modal.Footer>
      </Modal>
    </>
  );
};

export default LegaleseDialog;
