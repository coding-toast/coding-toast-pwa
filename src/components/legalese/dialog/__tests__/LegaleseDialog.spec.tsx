import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LegaleseDialog from '../LegaleseDialog';
import { Button, Modal } from 'react-bootstrap';

describe('<LegaleseDialog />', () => {
  it('renders correctly as link', () => {
    const component = shallow(<LegaleseDialog title='test title' />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders correctly with children', () => {
    const component = shallow(
      <LegaleseDialog title='test title'>
        <p>Hello test world!</p>
      </LegaleseDialog>
    );

    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders correctly as muted link', () => {
    const component = shallow(<LegaleseDialog title='test title' muted />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('open button shows the modal', () => {
    const component = shallow(<LegaleseDialog title='test title' />);

    const button = component.find(Button).at(0);
    button.simulate('click');

    const modal = component.find(Modal);

    expect(modal.props().show).toBeTruthy();
  });

  it('close button will hide the modal', () => {
    const component = shallow(<LegaleseDialog title='test title' />);

    const openButton = component.find(Button).at(0);
    openButton.simulate('click');

    const closeButton = component.find(Button).at(1);
    closeButton.simulate('click');

    const modal = component.find(Modal);

    expect(modal.props().show).toBeFalsy();
  });
});
