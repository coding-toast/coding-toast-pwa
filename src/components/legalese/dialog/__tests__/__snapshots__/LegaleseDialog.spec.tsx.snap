// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LegaleseDialog /> renders correctly as link 1`] = `
<Fragment>
  <Button
    active={false}
    className="p-0 "
    disabled={false}
    onClick={[Function]}
    size="sm"
    type="button"
    variant="link"
  >
    test title
  </Button>
  <Bootstrap(Modal)
    onHide={[Function]}
    show={false}
    size="lg"
  >
    <ModalHeader
      closeButton={true}
      closeLabel="Close"
    >
      <ModalTitle>
        test title
      </ModalTitle>
    </ModalHeader>
    <ModalBody>
      <ContractArea />
    </ModalBody>
    <ModalFooter>
      <Button
        active={false}
        block={true}
        className="btn-accent-secondary"
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="primary"
      >
        I understand
      </Button>
    </ModalFooter>
  </Bootstrap(Modal)>
</Fragment>
`;

exports[`<LegaleseDialog /> renders correctly as muted link 1`] = `
<Fragment>
  <Button
    active={false}
    className="p-0 text-muted"
    disabled={false}
    onClick={[Function]}
    size="sm"
    type="button"
    variant="link"
  >
    test title
  </Button>
  <Bootstrap(Modal)
    onHide={[Function]}
    show={false}
    size="lg"
  >
    <ModalHeader
      closeButton={true}
      closeLabel="Close"
    >
      <ModalTitle>
        test title
      </ModalTitle>
    </ModalHeader>
    <ModalBody>
      <ContractArea />
    </ModalBody>
    <ModalFooter>
      <Button
        active={false}
        block={true}
        className="btn-accent-secondary"
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="primary"
      >
        I understand
      </Button>
    </ModalFooter>
  </Bootstrap(Modal)>
</Fragment>
`;

exports[`<LegaleseDialog /> renders correctly with children 1`] = `
<Fragment>
  <Button
    active={false}
    className="p-0 "
    disabled={false}
    onClick={[Function]}
    size="sm"
    type="button"
    variant="link"
  >
    test title
  </Button>
  <Bootstrap(Modal)
    onHide={[Function]}
    show={false}
    size="lg"
  >
    <ModalHeader
      closeButton={true}
      closeLabel="Close"
    >
      <ModalTitle>
        test title
      </ModalTitle>
    </ModalHeader>
    <ModalBody>
      <ContractArea>
        <p>
          Hello test world!
        </p>
      </ContractArea>
    </ModalBody>
    <ModalFooter>
      <Button
        active={false}
        block={true}
        className="btn-accent-secondary"
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="primary"
      >
        I understand
      </Button>
    </ModalFooter>
  </Bootstrap(Modal)>
</Fragment>
`;
